asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
Thanks for the explanation @willnorris, that makes sense.

I agree that the `Do` and `NewRequest` separation makes more sense and is cleaner/safer, but I'd want to think more and understand it better (outside of this PR). I think we should consider changing this library in that direction, rather than the `Doer`, which I think is a lot harder for people to use (at all, and also correctly).

What are your thoughts on this @anpryl?